### PR TITLE
[DEVOPS #10] Add CI pipeline for data

### PR DESCRIPTION
**Pull request by labrdev**
### This Pull Request resolve the Devops Issue #10
This GitHub Actions YAML file named "CI Data" defines a workflow that is triggered when a pull request is made to the "main" branch and when changes are detected in files located in the "data" directory. The workflow has several key components.

The job named "ci" that runs on the "ubuntu-latest" environment. This job includes several steps:

1. "Checkout code": It checks out the repository code using the "actions/checkout@v2" action.
2. "Set up Python 3.11": It sets up Python 3.11 using the "actions/setup-python@v2" action, allowing the workflow to use the required Python version.
3. "Check Requirements file existence": It checks if the "data/requirements.txt" file exists using a custom action "thebinaryfelix/check-file-existence-action@1.0.0" and stores the result in an output variable.
4. "Cache dependencies": It caches Python dependencies in the "pip" cache directory, but only if the "requirements.txt" file exists. This helps speed up future workflow runs by reusing cached dependencies when the requirements file hasn't changed.
5. "Install dependencies": It installs or upgrades required Python packages, including "ruff," "pytest," "pynblint," and "lessc." It also checks if "requirements.txt" exists and installs its dependencies if present.
6. "Lint syntax errors with ruff": This step uses the "ruff" tool to check for Python syntax errors and undefined names in the code. It specifies certain formatting rules to ignore and targets Python version 3.11.
7. "Run Tests": This placeholder step is commented out but indicates the intention to run tests related to data science. The actual test execution is left to be defined or implemented later.

### CODE REVIEW

File: .github/workflows/data_ci.yaml [hunk 1]
```diff
@@ -0,0 +1,66 @@
+# Define the name of the GitHub Action workflow
+name: CI Data 
+
+# Trigger the workflow on pull requests to the 'main' branch
+on:
+  pull_request:
+    branches:
+      - main
```
Comment (eds000n) [inline .github/workflows/data_ci.yaml:8]: we need actions on `develop` branch as well

File: .github/workflows/data_ci.yaml [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+    # Define the steps to be executed within the job
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2  # Check out the repository code
+
+      - name: Set up Python 3.11
+        uses: actions/setup-python@v2
+        with:
+          python-version: '3.11'  # Use Python 3.11
```
Comment (eds000n) [inline .github/workflows/data_ci.yaml:31]: did you double check that the main libraries have support for 3.11?

Comment (eds000n) [reply .github/workflows/data_ci.yaml:31]: @unmsmfisi-socialapplication/data have  a look here pls

File: .github/workflows/data_ci.yaml (hunk 1 above)
Comment (labrdev) [reply .github/workflows/data_ci.yaml]: The change was apply. Now it's referencing to `develop`

File: .github/workflows/data_ci.yaml (hunk 2 above)
Comment (labrdev) [reply .github/workflows/data_ci.yaml]: By consulting with the team, it will change to 3.10

**Review by eds000n — APPROVED** [review]

**Pull request merged**